### PR TITLE
Added tests for NumberParser

### PR DESCRIPTION
**Pull request by rchakra3**
Added tests for the NumberParser class.

I've created a [tool](https://github.com/rchakra3/wolf) that uses Randoop to generate tests and then prunes them using [PIT](http://pitest.org) as a measure of quality. So far it's been useful in understanding which classes need more tests in a repo and approaches on how to add Line and Mutation Coverage. 

Thought I would start with one of the simpler classes to test and once I get this reviewed I can submit a few more PRs with tests for other classes as well!

### CODE REVIEW

**Comment by rchakra3** [comment]
Improvement in Test Coverage:

Before:
before_rome_np

After:
after_rome_np

File: rome/src/test/java/com/rometools/rome/io/impl/NumberParserTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    @Test
+    public void testLongParseFailure() {
+        Long num = NumberParser.parseLong("Non Long");
+        assertNull(num);
+    }
+
+    @Test
+    public void testLongParseSuccess() {
+        Long num = NumberParser.parseLong("1");
+        assertTrue(num == 1L);
```
Comment (mishako) [inline rome/src/test/java/com/rometools/rome/io/impl/NumberParserTest.java]: Would `assertEquals` work? Same for other tests.

File: rome/src/test/java/com/rometools/rome/io/impl/NumberParserTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Test
+    public void testLongParseDefaultSuccess() {
+        Long num = NumberParser.parseLong("Non Long", 1L);
+        assertTrue(num == 1L);
+
+        num = NumberParser.parseLong("0", 1L);
+        assertTrue(num == 0L);
```
Comment (mishako) [inline rome/src/test/java/com/rometools/rome/io/impl/NumberParserTest.java]: Two tests in one method? Same for `testFloatParseDefaultSuccess`.

File: rome/src/test/java/com/rometools/rome/io/impl/NumberParserTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Test
+    public void testLongParseSuccess() {
+        Long num = NumberParser.parseLong("1");
+        assertTrue(num == 1L);
+    }
+
+    @Test
+    public void testLongParseDefaultSuccess() {
+        Long num = NumberParser.parseLong("Non Long", 1L);
```
Comment (mishako) [inline rome/src/test/java/com/rometools/rome/io/impl/NumberParserTest.java]: This method returns primitive `long`, no need for boxing. Same for other methods like this.

File: rome/src/test/java/com/rometools/rome/io/impl/NumberParserTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        Integer num = NumberParser.parseInt("Non Int");
+        assertNull(num);
+    }
+
+    @Test
+    public void testIntParseSuccess() {
+        Integer num = NumberParser.parseInt("1");
+        assertTrue(num == 1);
+    }
+
```
Comment (mishako) [inline rome/src/test/java/com/rometools/rome/io/impl/NumberParserTest.java]: Unnecessary blank line.

**Comment by mishako** [comment]
Hi Rohan! Thank you for the contribution. Looks good! I left some comments. They might be a bit nitpicky, but since you're going to continue with this kind of pull requests, I think it's good to do everything properly from the start.

**Comment by rchakra3** [comment]
Thanks for the review @mishako ! Really appreciate it. I'll make the
changes and update soon!

**Comment by rchakra3** [comment]
@mishako I've made the suggested changes. I'd appreciate you taking a quick look!

**Comment by mishako** [comment]
@rchakra3 You've included commits from master in your branch.

**Comment by rchakra3** [comment]
@mishako Sorry I've fixed that!

**Comment by mishako** [comment]
@rchakra3 Nice! Do you want to squash commits into one before we merge it?

**Comment by rchakra3** [comment]
@mishako Done! :)

**Comment by mishako** [comment]
@rchakra3 Ah, sorry, I missed more things:
- The first argument of `assertEquals(a, b)` is expected value, not the other way around.
- The first argument of `assertEquals(a, b, c)` is message (see two bottom tests).

**Comment by rchakra3** [comment]
@mishako I've made the changes. Do the messages look fine?

**Comment by mishako** [comment]
@rchakra3 The messages look fine, but they are pretty much what assertEquals has by default, so do you really need them?

**Comment by rchakra3** [comment]
@mishako That's true. I assumed you expected a custom message as per your second point above:
"The first argument of assertEquals(a, b, c) is message (see two bottom tests)."
since assertEquals(a, b, c) with `a` as the message seems to be deprecated

Did I misunderstand you?

**Comment by mishako** [comment]
@rchakra3 No, it's me, sorry. You were correctly using delta as the third argument, I didn't know about it and assumed you made a mistake. Sorry! :)

**Comment by rchakra3** [comment]
@mishako Not a problem! :)

I think I've got all the changes in now!

**Pull request merged**